versioned/version0_3_0.ts

```diff
@@ -0,0 +1,530 @@
+                      },
+                    })
+                  ),
+                  lambda: michelsonCode,
+                },
+              };
+            }
+            default:
+              return {};
+          }
+        })
+      )
+      .toTransferParams();
+
+    let op = await t.wallet.transfer(params).send();
+
+    const transacValue = await promiseWithTimeout(
+      op.transactionOperation(),
+      DEFAULT_TIMEOUT
+    );
+
+    if (transacValue === -1) {
+      return [true, op.opHash];
+    }
+
+    const confirmationValue = await promiseWithTimeout(
+      op.confirmation(1),
+      DEFAULT_TIMEOUT
+    );
+
+    if (confirmationValue === -1) {
+      return [true, op.opHash];
+    }
+
+    return [false, op.opHash];
+  }
+  async signProposal(
+    cc: WalletContract,
+    t: TezosToolkit,
+    proposal: number,
+    result: boolean | undefined,
+    resolve: boolean
+  ): Promise<timeoutAndHash> {
+    const proposals: { proposals: BigMapAbstraction } = await cc.storage();
+    const b_proposal = num2PaddedHex(proposal);
+    const prop: any = await proposals.proposals.get(b_proposal);
+    const batch = t.wallet.batch();
+    if (typeof result != "undefined") {
+      await batch.withContractCall(
+        cc.methods.sign_proposal(b_proposal, prop.contents, result)
+      );
+    }
+    if (resolve) {
+      const schema = new Schema(proposals_type);
+      const proposals_data = schema.Encode(prop.contents);
+      const packed = packDataBytes(proposals_data, proposals_type).bytes;
+
+      await batch.withContractCall(
+        // resolve proposal
+        cc.methods.proof_of_event_challenge(b_proposal, packed)
+      );
+    }
+    let op = await batch.send();
+
+    const confirmationValue = await promiseWithTimeout(
+      op.confirmation(1),
+      DEFAULT_TIMEOUT
+    );
+
+    if (confirmationValue === -1) {
+      return [true, op.opHash];
+    }
+
+    return [false, op.opHash];
+  }
+
+  async submitSettingsProposals(
+    cc: Contract,
+    t: TezosToolkit,
+    ops: ownersForm[]
+  ): Promise<timeoutAndHash> {
+    let content = ops
+      .map(v => {
+        if ("addOwners" in v) {
+          return { add_owners: v.addOwners };
+        } else if ("removeOwners" in v) {
+          return { remove_owners: v.removeOwners };
+        } else if ("changeThreshold" in v) {
+          return { adjust_threshold: Number(v.changeThreshold) };
+        } else if ("adjustEffectivePeriod" in v) {
+          return { adjust_effective_period: v.adjustEffectivePeriod };
+        } else {
+          return v;
+        }
+      })
+      .filter(x => !!x);
+
+    let params = cc.methods.create_proposal(content).toTransferParams();
+
+    let op = await t.wallet.transfer(params).send();
+
+    const transacValue = await promiseWithTimeout(
+      op.transactionOperation(),
+      DEFAULT_TIMEOUT
+    );
+
+    if (transacValue === -1) {
+      return [true, op.opHash];
+    }
+
+    return [false, op.opHash];
```

Comment:
I actually wrote some pretty bad code as it comes from me 😅 Would you mind updating it here and on the other versions to
```js
return [transacValue === -1, op.opHash]
```

Reply:
You updated it in `0_1_1` but not here, can you do it as well ? Once it's done it should be good

Reply:
I modified in the different function in both file. now it all modified.